### ISSUE TITLE
Deprecate our magnetostatics functionality?

### ISSUE THREAD

**Issue by StanczakDominik** [PlasmaPy/PlasmaPy#1688]
Last telecon, myself, @rocco8773 and @namurphy discussed that our magnetostatics functionality, while okay-ish, is essentially a subset of functionality contained in [MagpyLib](https://magpylib.readthedocs.io/en/latest/). We'd like to trim some of the less-used and less maintained functionality the package , so that one seems like a prime candidate.

Things to consider:
- [ ] IIRC, MagpyLib gives outputs in mT. One thing we considered during the telecon was to monkeypatch/provide shims on top of their functionality that makes stuff return astropy Quantities.
- [ ] should we wire the existing `magnetostatics` functionality to stuff in MagpyLib, or just deprecate it and tell users to use MagpyLib directly instead?

**Comment by pheuer** [comment]
I agree - MagpyLib is quite good and being actively maintained (last release was only a couple weeks ago). I think it makes sense to make any contributions we want directly to MagpyLib rather than having our own version of essentially the same thing. 

I vote that the magnetostatics notebook remain and be reworked in to an example using MagpyLib and the Plasmapy Plasma class/CartesianGrid to show how to use MagpyLib with PlasmaPy. If that can be done easily enough, then the magnetosonics module is clearly superfluous.
